tests/Unit/CssInlinerTest.php

```diff
@@ -2422,160 +2094,6 @@ function (array $dataset) {
-     *
-     * @dataProvider documentTypeAndSelfClosingTagDataProvider
-     */
-    public function renderConvertsXmlSelfClosingTagsToNonXmlSelfClosingTag(
-        $documentType,
-        $htmlWithNonXmlSelfClosingTags,
-        $htmlWithXmlSelfClosingTags
-    ) {
-        $subject = $this->buildDebugSubject(
-            $documentType . '<html><body>' . $htmlWithXmlSelfClosingTags . '</body></html>'
-        );
-
-        $result = $subject->render();
-
-        self::assertContains('<body>' . $htmlWithNonXmlSelfClosingTags . '</body>', $result);
-    }
-
-    /**
-     * @test
-     *
-     * @param string $documentType
-     * @param string $htmlWithNonXmlSelfClosingTags
-     *
-     * @dataProvider documentTypeAndSelfClosingTagDataProvider
-     */
-    public function renderKeepsNonXmlSelfClosingTags($documentType, $htmlWithNonXmlSelfClosingTags)
-    {
-        $subject = $this->buildDebugSubject(
-            $documentType . '<html><body>' . $htmlWithNonXmlSelfClosingTags . '</body></html>'
-        );
-
-        $result = $subject->render();
-
-        self::assertContains('<body>' . $htmlWithNonXmlSelfClosingTags . '</body>', $result);
-    }
-
-    /**
-     * @test
-     *
-     * @param string $htmlWithNonXmlSelfClosingTags
-     * @param string $tagName
-     *
-     * @dataProvider nonXmlSelfClosingTagDataProvider
-     */
-    public function renderNotAddsClosingTagForSelfClosingTags($htmlWithNonXmlSelfClosingTags, $tagName)
-    {
-        $subject = $this->buildDebugSubject(
-            '<html><body>' . $htmlWithNonXmlSelfClosingTags . '</body></html>'
-        );
-
-        $result = $subject->render();
-
-        self::assertNotContains('</' . $tagName, $result);
-    }
-
-    /**
-     * @test
-     */
-    public function renderAutomaticallyClosesUnclosedTag()
-    {
-        $subject = $this->buildDebugSubject('<html><body><p></body></html>');
-
-        $result = $subject->render();
-
-        self::assertContains('<body><p></p></body>', $result);
-    }
-
-    /**
-     * @test
-     */
-    public function renderReturnsCompleteHtmlDocument()
-    {
-        $subject = $this->buildDebugSubject('<html><body><p></p></body></html>');
-
-        $result = $subject->render();
-
-        self::assertSame(
-            $this->html5DocumentType . "\n" .
-            "<html>\n" .
-            '<head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"></head>' . "\n" .
-            "<body><p></p></body>\n" .
-            "</html>\n",
-            $result
-        );
-    }
-
-    /**
-     * @test
-     */
-    public function renderBodyContentReturnsBodyContentFromHtml()
-    {
-        $subject = $this->buildDebugSubject('<html><body><p></p></body></html>');
-
-        $result = $subject->renderBodyContent();
-
-        self::assertSame('<p></p>', $result);
-    }
-
-    /**
-     * @test
-     */
-    public function renderBodyContentReturnsBodyContentFromPartialContent()
```

Comment:
We don't seem to have an equivalent test in `AbstractHtmlProcessorTest` so should this be moved there rather than removed?

Reply:
This is covered by several tests that test several smaller aspects of this:
-  `addsMissingHtmlTag`
- `addsMissingBodyTag'
- `renderBodyContentReturnsBodyContent`

Reply:
I see.